src/components/ValidateConfigs.ts

```diff
@@ -0,0 +1,119 @@
+            return `${widgetName}: supplied entity "${props.entity}" does not belong to list view data source reference`;
+        }
+        if (props.entity && ValidateConfigs.itContains(props.entity, "/")) {
+            // const entityPath = ValidateConfigs.getRelatedEntity(props);
+            // if (props.entity && !ValidateConfigs.isValidAttribute(entityPath, props)) {
+            //     // return `${widgetName}: supplied attribute name "${props.values.attribute}" does not belong to list view data source reference`;
+            //     return `${widgetName}: supplied attribute name does not belong to list view data source reference`;
+            // }
+        }
+
+        return "";
+    }
+
+    static getRelatedEntity(props: ValidateConfigProps): string {
+        if (props.targetGrid) {
+            const dataSourceEntity = window.mx.meta.getEntity(props.targetGrid._datasource._entity);
+            const referenceAttributes: string[] = dataSourceEntity.getReferenceAttributes();
+            for (const referenceAttribute of referenceAttributes) {
+                if (ValidateConfigs.itContains(props.entity, referenceAttribute)) {
+                    const selectorEntity = dataSourceEntity.getSelectorEntity(referenceAttribute);
+                    if (ValidateConfigs.itContains(props.entity, selectorEntity)) {
+                        return selectorEntity;
+                    }
+                }
+            }
+        }
+
+        return "";
+    }
+
+    // static isValidAttribute(entity: string, props: ValidateConfigProps): boolean {
+    //     if (props.targetGrid) {
+    //         const dataSourceEntity: mendix.lib.MxMetaObject = window.mx.meta.getEntity(entity);
+    //         const dataAttributes: string[] = dataSourceEntity.getAttributes();
+    //         if (ValidateConfigs.itContains(dataAttributes, props.values.)) {
+    //             return true;
+    //         }
+    //     }
+
+    //     return false;
+    // }
+
+    static isCompatible(targetGrid: ListView): boolean {
+        return !!(targetGrid &&
+            targetGrid._onLoad &&
+            targetGrid._loadMore &&
+            targetGrid._renderData &&
+            targetGrid._datasource &&
+            targetGrid._datasource.atEnd &&
+            typeof targetGrid._datasource._pageSize !== "undefined" &&
+            typeof targetGrid._datasource._setSize !== "undefined");
+    }
+
+    static findTargetNode(props: DropdownFilterContainerProps, filterNode: HTMLElement): HTMLElement | null {
+        let targetNode: HTMLElement | null = null ;
+
+        while (!targetNode && filterNode) {
+            targetNode = props.targetGridName
+                ? filterNode.querySelector(`.mx-name-${props.targetGridName}`) as HTMLElement
+                : filterNode.querySelectorAll(`.mx-listview`)[0] as HTMLElement;
+
+            if (targetNode) break;
```

Comment:
Single line if statements, please dont use the, and if you use them at least add the `{ }`

Reply:
Done